### PR TITLE
common: Add docs' stubs and scripts for manpages

### PR DESCRIPTION
**Pull request by lukaszstolarczuk**
updated structure of man pages, with approach not to commit generated files

---
This change is [Reviewable](https://reviewable.io/reviews/pmem/pmemkv/292)

### CODE REVIEW

File: doc/pmemkv/pmemkv.7.md [hunk 1]
```diff
@@ -0,0 +1,64 @@
+---
+layout: manual
+Content-Style: 'text/css'
+title: PMEMKV
+header: PMEMKV
+date: pmemkv version 0.8
+...
+
+[comment]: <> (Copyright 201(?)-2019, Intel Corporation)
```
Comment (marcinslusarz) [inline doc/pmemkv/pmemkv.7.md]: ? ?

File: utils/md2man/default.man [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+.\"t
+$endif$
+$if(pandoc-version)$
+.\" Automatically generated by Pandoc $pandoc-version$
+.\"
+$endif$
+$if(adjusting)$
+.ad $adjusting$
+$endif$
+.TH "$title$" "$section$" "$date$" "PMDK - $version$" "PMDK Programmer's Manual"
```
Comment (marcinslusarz) [inline utils/md2man/default.man]: PMDK?

File: CMakeLists.txt [hunk 3]
```diff
@@ -267,6 +267,12 @@ install(FILES ${CMAKE_BINARY_DIR}/libpmemkv.pc
 
 set_target_properties(pmemkv PROPERTIES PUBLIC_HEADER "src/libpmemkv.h;src/libpmemkv.hpp")
 
+add_custom_target(md2man
+	COMMAND ${CMAKE_SOURCE_DIR}/utils/md2man/md2man.sh
+	${CMAKE_SOURCE_DIR}/doc/pmemkv/pmemkv.7.md
+	${CMAKE_SOURCE_DIR}/utils/md2man/default.man
+	${CMAKE_SOURCE_DIR}/doc/generated/pmemkv.7)
```
Comment (marcinslusarz) [inline CMakeLists.txt]: this target should be run automatically whenever source md files are changed

File: doc/pmemkv/pmemkv.7.md [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+[NAME](#name)<br />
+[SYNOPSIS](#synopsis)<br />
+[DESCRIPTION](#description)<br />
+[EXAMPLE](#example)<br />
+[SEE ALSO](#see-also)<br />
+
+
+# NAME #
+
+**pmemkv** - Key/Value Datastore for Persistent Memory
```
Comment (marcinslusarz) [inline doc/pmemkv/pmemkv.7.md]: pmemkv or libpmemkv?

File: utils/md2man/md2man.sh [hunk 5]
```diff
@@ -0,0 +1,85 @@
...
+	-V year=$(date +"%Y") |
+sed '/^\.IP/{
+N
+/\n\.nf/{
+	s/IP/PP/
+    }
+}'
+
+# don't overwrite the output file if the only thing that changed
+# is modification date (diff output has exactly 4 lines in this case)
```
Comment (marcinslusarz) [inline utils/md2man/md2man.sh]: If we won't keep generated files in the repository this code is not needed

File: utils/md2man/md2man.sh [hunk 6]
```diff
@@ -0,0 +1,85 @@
...
+outfile=$3
+title=`sed -n 's/^title:\ _MP(*\([A-Za-z_-]*\).*$/\1/p' $filename`
+section=`sed -n 's/^title:.*\([0-9]\))$/\1/p' $filename`
+version=`sed -n 's/^date:\ *\(.*\)$/\1/p' $filename`
+
+dt=$(date +"%F")
+out_dir=`echo $outfile | sed 's/\(.*\)\/.*/\1/'`
+mkdir -p $out_dir
+cat $filename | sed -n -e '/# NAME #/,$p' |\
+	pandoc -s -t man -o $outfile.tmp --template=$template \
```
Comment (marcinslusarz) [inline utils/md2man/md2man.sh]: $outfile.tmp -> $outfile if you'll drop the code below

File: doc/pmemkv/pmemkv.7.md (hunk 1 above)
Comment (lukaszstolarczuk) [reply doc/pmemkv/pmemkv.7.md]: yeah, this date - I don't have an idea which year it should be? The first commit of pmemkv? 2019?

Comment (marcinslusarz) [reply doc/pmemkv/pmemkv.7.md]: This is a copyright for the man page, which is a new thing in this repo, so 2019 is enough.

**Comment by lukaszstolarczuk** [comment]
ready for review:
- stubs done
- generating manpages with pandoc
- readme updated

**Pull request merged**